### DISCUSSION TITLE
migration add enum operate

### DISCUSSION THREAD

**Discussion by caixingyue** [laravel/framework#51734] [Ideas] [answered]
It is recommended to add more enumeration-related functions in the enumeration field

For example, I often write like this:
`$table->enum('status', array_column(UserStatus::cases(), 'value'))->default(UserStatus::Normal)->comment('Account Status');`

Is there a better way to omit `array_column(UserStatus::cases(), 'value'),` for example, I can directly `$table->enum('status', UserStatus::class)->default(UserStatus::Normal)->comment('Account Status');`

**Comment by henzeb** [comment, marked as answer]
It is not recommended. If you change things or remove things, your migrations will fail...

**Comment by wizzymore** [comment]
I agree with @henzeb, make it weakly typed ( to reflect only the type of the BackedEnum ) and then simply cast it in the model so you can't insert bad data there.
